### PR TITLE
Inline the synchrotron and pusher configs into the main config

### PR DESCRIPTION
**Pull request by NegativeMjark**
The config will look something like:

```
worker_app: "synapse.app.pusher"
worker_daemonize: true
worker_pid_file: "pusher.pid"
worker_log_file: "pusher.log"
worker_replication_url: "http://127.0.0.1/_synapse/replication"
worker_listeners:
      - type: "manhole"
        port: 12345
      - type: "http"
        resources: [{names: ["metrics"]}]
        port: 54321
```

### CODE REVIEW

File: synapse/config/workers.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "gc_thresholds",
+    "replication_url",
+])
+
+
+def clobber_with_worker_config(config, worker_config):
+    """Overrides some of the keys of the main config with worker-specific
+    values."""
+    config.event_cache_size = worker_config.event_cache_size
+    config.replication_url = worker_config.replication_url
```
Comment (erikjohnston) [inline synapse/config/workers.py]: Why aren't we clobbering all values?

**Comment by erikjohnston** [comment]
`clobber_with_worker_config` feels a bit fragile to me. For example, if someone pulls out a config option deep within synapse it isn't obvious that it needs to be added to that function, and the only observable effect would be that overriding it in the worker config would silently not work.

**Comment by NegativeMjark** [comment]
> clobber_with_worker_config feels a bit fragile to me

I've replaced it with code that accesses the worker config directly bde1383 364d616

@erikjohnston PTAL

**Comment by NegativeMjark** [comment]
So rather than folding all the worker configs into a single dictionary, I've added "worker_" prefixes to all the config keys so that the existing mechanisms for merging multiple config files can be used.

@erikjohnston PTAL

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**